sources/api/ecs-settings-applier/src/ecs.rs

```diff
@@ -45,6 +45,12 @@ struct ECSConfig {
 
     #[serde(rename = "OverrideAWSLogsExecutionRole")]
     override_awslogs_execution_role: bool,
+
+    #[serde(
+        rename = "SpotInstanceDrainingEnabled",
+        skip_serializing_if = "Option::is_none"
+    )]
+    enable_spot_instance_draining: Option<bool>,
```

Comment:
This struct should match the generated JSON, so we can rely on the default serde transformation except for the cases where the expected JSON doesn't match that.  (The ECS agent generally follows the [Go convention for initialisms](https://github.com/golang/go/wiki/CodeReviewComments#initialisms) that I don't believe serde knows how to handle.)

```suggestion
    #[serde(skip_serializing_if = "Option::is_none")]
    spot_instance_draining_enabled: Option<bool>,
```